src/object-store/src/shared_realm.cpp

```diff
@@ -318,6 +320,13 @@ void Realm::verify_in_write() const
     }
 }
 
+void Realm::verify_open() const
+{
+    if (is_closed()) {
+        throw InvalidTransactionException("Can't read a Realm that was closed.");
```

Comment:
The comments in the header say that continuing to call methods on `Realm` after it is closed is undefined behavior. If we're going to change this policy, and adding checks of this nature feels like a change, I think we should do so consistently on public entrypoints of this class.

Since `begin_transaction` calls `read_group` without first checking `is_closed`, this could result in beginning a write transaction throwing an exception about reading a Realm. That seems like it could be confusing?